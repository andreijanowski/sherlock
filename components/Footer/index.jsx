import { PureComponent } from "react";
import withI18next from "lib/withI18next";
import { func } from "prop-types";
import { Flex } from "@rebass/grid";

const namespaces = ["common"];

class Footer extends PureComponent {
  render() {
    const { t } = this.props;
    return <Flex>{t("footer")}</Flex>;
  }
}

Footer.propTypes = {
  t: func.isRequired
};

export default withI18next(namespaces)(Footer);
