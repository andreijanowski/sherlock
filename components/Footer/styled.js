import styled from "styled-components";
import { Flex } from "@rebass/grid";

import { WRAPPER_WIDTH } from "sections/common/sharedStyled";

export const FooterWrapper = styled(Flex).attrs(() => ({
  as: "footer",
  width: 1,
  flexDirection: "column",
  pb: 20,
  px: 3,
  alignItems: "flex-start"
}))`
  color: rgb(${p => p.theme.colors.white});
  max-width: ${WRAPPER_WIDTH}px;
  margin: auto;
`;

export const MenuWrapper = styled(Flex).attrs(() => ({
  alignSelf: "center",
  justifyContent: "flex-start",
  flexWrap: "wrap",
  width: 1
}))`
  flex: 1;
  max-width: 1150px;
`;

export const NavItem = styled(Flex).attrs(() => ({
  flexDirection: "column",
  justifyContent: "flex-start"
}))``;

export const ColumnTitle = styled.p`
  margin-bottom: 16px;
  font-weight: ${p => p.theme.fontWeights.bold};
  font-size: ${p => p.theme.fontSizes.f18};
  line-height: ${p => p.theme.fontSizes.f27};
  letter-spacing: 0.7px;
  color: rgb(${p => p.theme.colors.white});
`;

export const ListWrapper = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
`;

export const NavigationLink = styled.a`
  display: block;
  margin-bottom: 4px;
  font-weight: ${p => p.theme.fontWeights.medium};
  font-size: ${p => p.theme.fontSizes.f16};
  line-height: ${p => p.theme.fontSizes.f24};
  color: rgb(${p => p.theme.colors.white});
  text-decoration: none;
  &:hover {
    color: rgba(${p => p.theme.colors.white}, 0.7);
  }
`;

export const Line = styled.hr`
  width: 100%;
  margin: 70px 0 20px;
  border: none;
  border-bottom: 1px solid rgb(${p => p.theme.colors.gray["4"]});
`;

export const SocialsContainer = styled(Flex).attrs({
  as: "ul",
  alignItems: "center",
  flexWrap: "nowrap"
})`
  margin: 0;
  padding: 0;
  list-style: none;
`;

export const SocialsItem = styled.li`
  &:not(:last-child) {
    margin-right: 16px;
  }
`;

export const SocialsItemLink = styled.a``;

export const LogoContainer = styled.div`
  color: rgb(${p => p.theme.colors.white});
  font-weight: ${p => p.theme.fontWeights.semiBold};
  font-size: ${p => p.theme.fontSizes.f14};
  line-height: ${p => p.theme.fontSizes.f18};
  letter-spacing: 0.5px;
`;

export const LogoSmallText = styled.div`
  font-weight: ${p => p.theme.fontWeights.medium};
`;

export const CopyrightsContainer = styled.div`
  color: rgb(${p => p.theme.colors.white});
  font-weight: ${p => p.theme.fontWeights.thin};
  font-size: ${p => p.theme.fontSizes.f12};
  line-height: ${p => p.theme.fontSizes.f18};
  letter-spacing: 0.7px;
`;
