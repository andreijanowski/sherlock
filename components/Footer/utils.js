import {
  apiGuideLink,
  becomePartnerLink,
  mediaKitLink,
  privacyPolicyLink,
  termsAndConditionsLink,
  INVEST_TYPEFORM_LINK,
  LICENSING_TYPEFORM_LINK
} from "consts";

export const getConfig = t => [
  {
    title: t("navigation.columnOne.title"),
    name: "columnOne",
    hrefs: {
      0: {
        href: becomePartnerLink,
        target: "_blank",
        rel: "noreferrer noopener"
      },
      1: {
        href: `mailto:hello@foodetective.co?subject=${t("workInFoodetective")}`,
        target: "_blank",
        rel: "noreferrer noopener"
      },
      2: {
        href: LICENSING_TYPEFORM_LINK,
        target: "_blank",
        rel: "noreferrer noopener"
      },
      3: {
        href: INVEST_TYPEFORM_LINK,
        target: "_blank",
        rel: "noreferrer noopener"
      }
    }
  },
  {
    title: t("navigation.columnTwo.title"),
    name: "columnTwo",
    hrefs: {
      0: {
        href: "https://medium.com/@foodetective",
        target: "_blank",
        rel: "noreferrer noopener"
      },
      1: {
        href: mediaKitLink,
        target: "_blank",
        rel: "noreferrer noopener"
      },
      2: {
        href: apiGuideLink,
        target: "_blank",
        rel: "noreferrer noopener"
      }
    }
  },
  {
    title: t("navigation.columnThree.title"),
    name: "columnThree",
    hrefs: {
      0: {
        href: privacyPolicyLink,
        target: "_blank",
        rel: "noreferrer noopener"
      },
      1: {
        href: termsAndConditionsLink,
        target: "_blank",
        rel: "noreferrer noopener"
      }
    }
  }
];

export const socialsConfig = [
  {
    icon: "/static/img/socials/fb.svg",
    href: "https://www.facebook.com/business.foodetective/"
  },
  {
    icon: "/static/img/socials/in.svg",
    href: "https://www.linkedin.com/company/13981713/"
  },
  {
    icon: "/static/img/socials/tw.svg",
    href: "https://twitter.com/FoodetectivePro"
  },
  {
    icon: "/static/img/socials/ig.svg",
    href: "https://www.instagram.com/business.foodetective.co/"
  },
  {
    icon: "/static/img/socials/yt.svg",
    href: "https://www.youtube.com/channel/UC2MnDoNtpuh5CTbhEtOOGbA/featured"
  }
];
