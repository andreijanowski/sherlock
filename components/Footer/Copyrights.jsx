import React from "react";

import { FOODETECTIVE_ADDRESS } from "consts";
import { CopyrightsContainer } from "./styled";

const Logo = () => {
  const year = new Date().getFullYear();
  return (
    <CopyrightsContainer>
      {`Copyright Â© ${year} Foodetective. All rights reserved.`}
      <br />
      {FOODETECTIVE_ADDRESS}
    </CopyrightsContainer>
  );
};

export default Logo;
