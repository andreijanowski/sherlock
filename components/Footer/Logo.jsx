import React from "react";
import { string } from "prop-types";

import { Link } from "components";
import { LogoContainer, LogoSmallText } from "./styled";

const Logo = ({ lng }) => (
  <Link route="/" lng={lng}>
    <LogoContainer as="a">
      Foodetective <LogoSmallText> For Business</LogoSmallText>
    </LogoContainer>
  </Link>
);

Logo.propTypes = {
  lng: string.isRequired
};

export default Logo;
