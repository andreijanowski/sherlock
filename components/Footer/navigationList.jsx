import { func, string, shape } from "prop-types";
import { ListWrapper, NavigationLink } from "./styled";

const navigationList = ({ t, columnName, hrefs }) => {
  const listLength = t(`navigation.${columnName}.links.length`);
  const linksList = [];
  for (let i = 0; i < listLength; i += 1) {
    const linkName = t(`navigation.${columnName}.links.${i}`);
    linksList.push(
      <li key={linkName}>
        <NavigationLink href={hrefs ? hrefs[i] : "#"}>
          {linkName}
        </NavigationLink>
      </li>
    );
  }
  return (
    <nav>
      <ListWrapper>{linksList}</ListWrapper>
    </nav>
  );
};

navigationList.propTypes = {
  t: func.isRequired,
  columnName: string.isRequired,
  hrefs: shape()
};

navigationList.defaultProps = {
  hrefs: null
};

export default navigationList;
