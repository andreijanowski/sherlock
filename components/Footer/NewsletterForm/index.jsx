import React, { useCallback, useState } from "react";
import { isEmail } from "validator";

import { client } from "sagas/api";
import { contentTypes } from "consts";
import { useT } from "utils/hooks";
import { LoadingIndicator } from "components";
import {
  StyledButton,
  Caption,
  Container,
  Error,
  Input,
  InputWrapper,
  LoaderWrapper
} from "./styled";

const STATUSES = {
  IDLE: "idle",
  LOADING: "loading",
  SUCCESS: "success",
  FAIL: "fail"
};

const NewsletterForm = () => {
  const t = useT("footer");
  const [status, setStatus] = useState(STATUSES.IDLE);
  const [inputValue, setInputValue] = useState("");

  const onSendCLick = useCallback(async () => {
    setStatus(STATUSES.LOADING);
    try {
      const {
        data: { success }
      } = await client.post(
        "/newsletter-subscription",
        {
          email: inputValue
        },
        {
          headers: {
            "Content-Type": contentTypes.JSON
          }
        }
      );
      setStatus(success ? STATUSES.SUCCESS : STATUSES.FAIL);
    } catch (e) {
      setStatus(STATUSES.FAIL);
    }
  }, [inputValue]);

  const onInputChange = useCallback(e => {
    setInputValue(e.target.value);
  }, []);

  const isEmailInvalid = inputValue ? !isEmail(inputValue) : false;
  const isSendDisabled = !inputValue || isEmailInvalid;

  return (
    <Container>
      {status === STATUSES.SUCCESS ? (
        <Caption>{t("newsletter.success")}</Caption>
      ) : (
        <>
          <InputWrapper isInvalid={isEmailInvalid} mb={2} pb="14px">
            <Input
              type="text"
              value={inputValue}
              onChange={onInputChange}
              placeholder={t("newsletter.email")}
            />
            <StyledButton
              disabled={isSendDisabled}
              onClick={onSendCLick}
              ml={2}
            >
              {t("newsletter.send")}
            </StyledButton>
          </InputWrapper>
          <Caption>{t("newsletter.caption")}</Caption>
          {status === STATUSES.FAIL && <Error>{t("newsletter.fail")}</Error>}
        </>
      )}
      {status === STATUSES.LOADING && (
        <LoaderWrapper>
          <LoadingIndicator isWrapped hasTransparentBackground />
        </LoaderWrapper>
      )}
    </Container>
  );
};

export default NewsletterForm;
