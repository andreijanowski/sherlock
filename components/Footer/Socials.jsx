import React from "react";

import { socialsConfig } from "./utils";
import { SocialsContainer, SocialsItem, SocialsItemLink } from "./styled";

const Socials = () => (
  <SocialsContainer>
    {socialsConfig.map(({ icon, href }) => (
      <SocialsItem key={href}>
        <SocialsItemLink href={href} target="_blank" rel="noreferrer noopener">
          <img src={icon} alt="" />
        </SocialsItemLink>
      </SocialsItem>
    ))}
  </SocialsContainer>
);

export default Socials;
