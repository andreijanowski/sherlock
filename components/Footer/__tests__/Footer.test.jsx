/* eslint-env jest */
import React from "react";
import { shallow } from "enzyme";
import Footer from "components/Footer";
import "jest-styled-components";

const props = {
  t: m => m
};

describe("Footer component", () => {
  const comp = <Footer {...props} />;
  const shallowComponent = shallow(comp);

  it("renders correctly", () => {
    expect(shallowComponent).toMatchSnapshot();
  });
});
