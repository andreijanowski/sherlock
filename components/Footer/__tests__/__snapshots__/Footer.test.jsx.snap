// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly 1`] = `
<I18n
  bindI18n="languageChanged loaded"
  bindStore="added removed"
  ns={
    Array [
      "common",
    ]
  }
  nsMode="default"
  omitBoundRerender={true}
  t={[Function]}
  translateFuncName="t"
  usePureComponent={false}
  withRef={false}
>
  <Component />
</I18n>
`;
